recursion.tex

```diff
@@ -228,23 +247,24 @@ \subsection{Implementing recursive queries}\label{sec:datalog}
 %$\code{O} = \fix{x}{R(\code{I}, x)}$.
 %\end{proof}
 
-Note that this loop is not guaranteed to terminate.  But if a program
-does converge, the above construction will find the least fixed point.
+Note that this loop is not guaranteed to terminate.  But if the
+program does converge, the above construction will find the least
+fixed point that contains $x$.
 
 In fact, this circuit implements the standard Datalog
 \defined{na\"{\i}ve evaluation} algorithm (e.g., see Algorithm~1 in
 \cite{greco-sldm15}).  Notice that the inner part of the circuit is
 the incremental form of another circuit, since it is sandwiched
-between $\I$ and $\D$ operators.  Using the cycle rule of we can
+between $\I$ and $\D$ operators.  Using the cycle rule we can
 rewrite this circuit:
```

Comment:
I bet that at this point most readers will miss the fact that the input to this circuit is not a stream (the input is a single arrow), so it's worth saying explicitly as the last sentence of the section that this circuit takes a DB snapshot and computes a recursive query in one shot.